node-wasm/src/ports.rs

```diff
@@ -64,6 +53,16 @@ pub(crate) struct Port {
     onmessage: Closure<dyn Fn(MessageEvent)>,
 }
 
+impl MessageId {
+    /// i++
+    pub(crate) fn post_increment(&mut self) -> MessageId {
+        let ret = *self;
+        let (next, _carry) = self.0.overflowing_add(1);
+        self.0 = next;
+        ret
+    }
+}
+
 impl Port {
     /// Create a new Port out of JS object, registering on message callback.
     /// Minimal duck-type checking is performed using reflection when setting appropriate properties.
```

Comment:
I think it could also take `MessagePortLike` instead of `JsValue`

Reply:
I believe original idea was to differenciate between Objects and Objects that we've checked that they are ports (since types here offer little safety, you can just JsValue.into() MessagePortLike). Here we get JsValue, then check with reflection appropriate properties and declare it port-like by returning MessagePortLike in `register_on_message_callback`.
We weren't that meticulous with upholding this convention, so I guess we can treat MessagePortLike as Object that we heavily wish was a port.